### DISCUSSION TITLE
Can this be used with electronjs?

### DISCUSSION THREAD

**Discussion by jijojosephk** [webauthn-open-source/fido2-lib#128] [Q&A]
Hi,

Can this be used with electron js to authenticate using USB security key?

**Comment by jijojosephk** [comment]
We'd like to address this issue here

https://github.com/IsmaelMartinez/teams-for-linux/issues/802

**Comment by JamesCullum** [comment]
I'm not sure if I know exactly what you mean - generally this library supports browser crypto as well, so it generally should work. Have we tested your specific setup and are aware of all electron-specific kinks? Probably not.

**Comment by jijojosephk** [comment]
I wanted to implement this.

https://developer.mozilla.org/en-US/docs/Web/API/CredentialsContainer/get

Electron apps does not have navigator.credentials.get implementation only google-chrome has. I really don't know how the security keys work. We would like to

1. Handle navigator.credentials.get and forward the request to the device.
2. User will have a key connected to the USB port.

**Comment by JamesCullum** [comment]
Thanks for the hint - if electron doesn't support the JS interface to even request a token, the backend cannot validate it. Keep in mind that this library is for validating the tokens exchanged - if it's not raised, it can't be validated.

So for now you'll have to address the electronjs devs to make this possible, eg [here](https://github.com/electron/electron/issues/24573)

**Reply by jijojosephk** [reply]
I can override `navigator.credentials.get` and added a console.log. This is what I get
```json
[
    {
        "publicKey": {
            "challenge": {},
            "timeout": 600000,
            "rpId": "login.microsoft.com",
            "allowCredentials": [],
            "userVerification": "required"
        }
    }
]
```
Is this something can be used to validate?

**Reply by JamesCullum** [reply]
This is the request from the website to the browser to do cryptographic things - it doesn't really work in this scenario. Of course you could override the methods and add your implementation of the browser crypto, but I'm not sure if that's a good perspective considering the time effort and the security implications.
